scenarioo-client/app/step/step.controller.js

```diff
@@ -17,34 +17,48 @@
 
 angular.module('scenarioo.controllers').controller('StepController', StepController);
 
-function StepController($scope, $routeParams, $location, StepResource, HostnameAndPort, SelectedBranchAndBuildService,
+function StepController($scope, $routeParams, $location, $route, StepResource, HostnameAndPort, SelectedBranchAndBuildService,
                         $filter, ApplicationInfoPopupService, GlobalHotkeysService, LabelConfigurationsResource,
                         SharePageService, SketcherContextService, RelatedIssueResource, SketchIdsResource,
-                        SketcherLinkService) {
+                        SketcherLinkService, BranchesAndBuildsService, ScreenshotUrlService, SelectedComparison, BuildDiffInfoResource,
+                        StepDiffInfoResource, DiffInfoService, localStorageService) {
 
     var transformMetadataToTreeArray = $filter('scMetadataTreeListCreator');
     var transformMetadataToTree = $filter('scMetadataTreeCreator');
 
     var selectedBranchAndBuild = {};
+    var selectedComparison;
     var useCaseName = $routeParams.useCaseName;
     var scenarioName = $routeParams.scenarioName;
+    var labels = $location.search().labels;
+
     $scope.pageName = $routeParams.pageName;
     $scope.pageOccurrence = parseInt($routeParams.pageOccurrence, 10);
     $scope.stepInPageOccurrence = parseInt($routeParams.stepInPageOccurrence, 10);
-    var labels = $location.search().labels;
-
+    $scope.comparisonInfo = SelectedComparison.info;
+    $scope.activeTab = getActiveTab();
+
+    $scope.displayDiffScreenshotKey = 'displayDiffScreenshot';
+    $scope.displayComparisonScreenshotKey = 'displayComparisonScreenshot';
+    $scope.displaySideBySideViewKey = 'displaySideBySideView';
+    $scope.displaySinglePageViewKey = 'displaySinglePageView';
```

Comment:
You can keep it. But I not agree that this is how we should design controllers.
Controller should define a nice interface for keeping the view as simple as possible.
I not consider the following view code nice and clean:
```
 <div id="sc-step-comparison-buttons" class="btn-group" ng-init="initStorageKeys()">
                        <button type="button" id="sc-side-by-side-view-button" ng-class="{active: comparisonViewOptions[displaySideBySideViewKey]}" ng-click="setActiveView(displaySideBySideViewKey)" class="btn btn-default"> <img ng-src="images/diffViewer/SideBySideView.png" class="sc-step-comparison-icon"/>Side by side view</button>
                        <button type="button" id="sc-single-page-view-button" ng-class="{active: comparisonViewOptions[displaySinglePageViewKey]}" ng-click="setActiveView(displaySinglePageViewKey)" class="btn btn-default" ><img ng-src="images/diffViewer/SinglePageView.png" class="sc-step-comparison-icon"/>Single page view</button>
                    </div>
```
and by the way, now I see that there is even a strange use of `ng-init`, which should be avoided, see https://docs.angularjs.org/api/ng/directive/ngInit

Initialization should be done in general in the controller, e.g. in the activate method.